rfc/rfc-000-lightweightRFCprocess.md

```diff
@@ -0,0 +1,76 @@
+# RFC-000: Lightweight RFC Process for riff
+
+**To be Reviewed By:** Nov 15th 2019
+
+**Authors:** Swapnil Bawaskar
+
+**Status:** **Active** | Accepted | Dropped | Superseded
+
+**Pull Request URL:** https://github.com/projectriff/riff/pull/1358
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** N/A
+
+
+## Problem
+As the scope of the project expands and the number of moving pieces increases, we need a process to gather feedback on the changes proposed to the architecture and interaction of various components in the system.
+
+This document will refer to that process as the “Request For Comments (RFC) process”.
+
+## Solution
+The proposed solution to address the problem described above is to have an individual author or a group of authors submit a proposal to the community in order to gather feedback and achieve consensus. The RFC follows the same format as used by this proposal.
+
+Much inspiration for this proposal has been drawn from [Apache Geode's RFC process](https://cwiki.apache.org/confluence/display/GEODE/Lightweight+RFC+Process), which in turn draws from Phil Calçado’s [Structured RFC Process](https://philcalcado.com/2018/11/19/a_structured_rfc_process.html).
+
+All RFCs are submitted via PRs to the [github.com/projectriff/riff](https://github.com/projectriff/riff) repo which will be merged when:
+- The `Reviewed By` date has elapsed
+- The RFC author has summarized the discussion in the PR thread.
+- concensus has been reached about the state ("Accepted/Dropped").
+
+### Collaboration
+Comments and feedback should be provided on the PR.
+
+Authors should address all comments by the deadline. This doesn't mean every comment and suggestion must be accepted and incorporated, but they should be carefully read and responded to.
+
+Every RFC is in one of the following phases:
+* **Active**: The RFC has been drafted and is open for comments. When making the PR, the doc should start out in this state.
```

Comment:
is Active the phase when the PR creating the RFC is open, or is there some periods after the proposal PR is open before it moves to Accepted?

For example, should Superseded By be updated as part of the initial PR, or is that something that happens once the RFC is accepted? Can a RFC be superseded by a dropped RFC?

I'd suggest dropped the Accepted status. The RFC being merged is an indication of acceptance, unless explicitly marked as Dropped.

Reply:
Marking the status explicitly as accepted before merging would help casual readers understand the status better.

Reply:
The PR gets created in Active state, and it would be better to explicitly move the state to "Accepted/Rejected" before merging.

Reply:
In that case, I'd suggest that all Active PRs be merged as is as an indication that the proposal is complete. Then a subsequent PR be opened as a yey/ney vote on the proposal.

Reply:
I like that idea (but that makes "Pending" seem even better than "Active" IMO), and the subsequent PR would be at a minimum just changing the status word itself.

Reply:
I think we should strive to arrive at a consensus in the original PR, rather than a voting PR later on. If there is indeed a need for a vote, that can take place in the original PR itself. (less things to link from the rfc doc).

Reply:
I am inclined to remove *Active* and only leave `Accepted | Rejected | Deprecated` as the possible states. PR starts off with empty state.

Reply:
That sounds good to me @sbawaska. Maybe eventually we'll want to add Deferred, but let's wait for a case.

Reply:
changed! https://github.com/projectriff/riff/pull/1358/commits/082d2fe2682668361eb6ab04c3b14cf7f0ea4633